adapters/repos/db/aggregator/iterator.go

```diff
@@ -0,0 +1,247 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package aggregator
+
+import (
+	"bytes"
+	"runtime"
+
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/sroar"
+	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+)
+
+func iteratorConcurrentlyReplace(b *lsmkv.Bucket, aggregateFunc func(b []byte) error, logger logrus.FieldLogger) error {
```

Comment:
I would put the iterator methods as part of each strategy struct not as separate methods

Reply:
not entirely sure where you want this - this probably cannot be reused as the function is different each time :/

Reply:
I wonder if generics would help get rid of code duplication

Reply:
don't think I can simplify that with generics, although I don't know it that well. I introduced a cursor interface to simplify things a little bit - let me know what you prefer